uwblueprint/sistema
# David/gcal export

# Notion ticket link

[GCAL Export](https://www.notion.so/uwblueprintexecs/GCAL-Export-7ef8b23d5057477385eb09f794e197f9?pvs=4)

## Implementation description
- Create a NextJS Route to extract absence information 
- Processed extracted data into downloadable ics file via the ics library
- Created seperate page to handle the downloads of ics files

## Steps to test

1. Run git pull on this branch to receive the updated docker files with the ability to query database
2. Run npm i to install necessary dependencies
3. Run the docker containers via docker desktop
4. Run npx prisma migrate dev --name init
5. Run npx @snaplet/seed init , say no to AI features
6. Run npx prisma db seed to seed the database
7. Query via docker and PSQL to check to ensure that database is seeded
8. Go to localhost:3000/ics and download the ics file
9. Make a separate calendar in google calendar and input the file into the calendar via the import & export function under the settings icon, multiple events should be exported

## What should reviewers focus on?
- Readability of Code, simplification of certain processes
- Any improvements to be made via db querying and file downloading process
- Adding comments to parts of the code that are hard to understand

## Checklist
